src/motor-operations.h

```diff
@@ -62,22 +74,32 @@ class MotorOperations {  // Rename SegmentQueue ?
 
   // Wait, until all elements in the ring-buffer are consumed.
   virtual void WaitQueueEmpty() = 0;
+
+  // Get the absolute position and auxes status currently
+  // executed by the backend.
+  virtual void GetRealtimeStatus(RealtimeStatus *status) = 0;
 };
 
+struct HistorySegment;
```

Comment:
Is history segment something that should be in the toplevel namespace ? It looks like it is something that should be a private type member of MotionQueueMotorOperations.
You can just forward declare it after the `private:`.